docs/src/pages/guides/publish-to-npm.md

```diff
@@ -1,101 +1,180 @@
 
-```astro
----
-import Capitalize from '@example/my-components/Capitalize.astro';
----
-<Capitalize phrase={`Hello world`} />
+```diff
+{
+  "name": "@example/my-package",
+  "version": "1.0.0",
+- "exports": "./index.js"
++ "exports": {
++    "./Bold.astro": "./Bold.astro",
++    "./Capitalize.astro": "./Capitalize.astro"
++ }
+}
 ```
 
-This is a less common scenario, and we only recommend it if you have good reason. Because Astro is completely rendered at build-time, there are no client-side performance concerns to our default recommendation to export your components from a single `index.js` file.
+## Developing your package
 
-To support importing by file within your package, add each file to your **package.json** `exports` map:
+Astro does not have a dedicated "package mode" for development. Instead, you should use a demo project to develop and test your package inside of the repo. This can be a private website only used for development, or a public demo/documentation website for your package.
+
+If you are extracting components from an existing project, you can even continue to use that project to develop your now-extracted components.
+
+For development, we **strongly recommend** that you setup a [**project workspace.**][Node-Packages-Workspace] Workspaces are a feature supported by all package managers including npm (v7+), yarn, pnpm and others. A workspace will help you verify package.json configuration and also let you import your package by name:
+
+```js
+// ✅ With workspaces:
+import {Bold, Capitalize} from 'package-name';
+
+// ❌ Without workspaces:
+import {Bold, Capitalize} from '../../../my-package-directory/index.js';
+```
+
+To turn your project into a workspace, create a `package.json` file at the top-most level of your repository (if one doesn't already exist) and add  a `"workspaces"` entry that identifies your `packages/` directory as a collection of packages:
 
 ```diff
 {
-  "name": "@example/my-components",
-  "version": "1.0.0",
-  "exports": {
--    ".": "./index.js",
-+    "./Bold.astro": "./Bold.astro",
-+    "./Capitalize.astro": "./Capitalize.astro"
-  }
+  "name": "my-workspace",
+  "Description": "this is the package.json that sits at the top-level of your repo.",
++ "workspaces": ["./packages/*"]
+}
+```
+
+Save this change, and re-run your package manager's "install" command to set up your new workspace. If everything went well, you can now import your workspace packages by name.
+
+You can optionally choose to move your development website's `src/` and `public/` directories into a workspace directory of their own. This is optional, and only recommended if you are familiar with how workspaces work.
+
+```diff
+{
+  "name": "my-workspace",
+  "Description": "this is the package.json that sits at the top-level of your repo.",
+  "workspaces": [
+    "./packages/*",
++   "./demo"
+  ]
 }
 ```
 
-## Community components
+## Testing your component
```

Comment:
I would want to more clearly explain the structure of testing like this... should the expected HTML go under `expected/__fixtures__/`?

Reply:
How would you feel about removing this section for now and tackling as a separate PR? Since this is totally new content, I don't think this PR would take away from that work.

Reply:
I think I'd rather see this go out immediately, and then have a second PR fully flesh out the story, maybe even with its own half/full page..

Reply:
sounds good to me! Yea, I'm fine with either. As mentioned on Discord, this will probably get rewritten when the new compiler is out, so what we have here may also be good enough until then.